cmd/root.go

```diff
@@ -220,14 +224,31 @@ func initConfigWithViperCtx(c *config) error {
 		}
 	}
 
+	certificateAuthorityData := viper.GetString("certificate-authority-data")
+
 	var (
-		client metalgo.Client
-		err    error
+		client    metalgo.Client
+		err       error
+		transport *http.Transport
 	)
-	if hmacAuthType != "" {
-		client, err = metalgo.NewDriver(driverURL, apiToken, hmacKey, metalgo.AuthType(hmacAuthType))
+
+	if certificateAuthorityData == "" {
+		if hmacAuthType == "" {
+			client, err = metalgo.NewClient(driverURL, metalgo.BearerToken(apiToken), metalgo.HMACAuth(hmacKey, "Metal-Admin"))
+		} else {
+			client, err = metalgo.NewClient(driverURL, metalgo.BearerToken(apiToken), metalgo.HMACAuth(hmacKey, hmacAuthType))
+		}
 	} else {
-		client, err = metalgo.NewDriver(driverURL, apiToken, hmacKey)
+		transport, err = createTLSTransport(certificateAuthorityData)
+		if err != nil {
+			return err
+		}
+
+		if hmacAuthType == "" {
+			client, err = metalgo.NewClient(driverURL, metalgo.BearerToken(apiToken), metalgo.HMACAuth(hmacKey, "Metal-Admin"), metalgo.Transport(transport))
+		} else {
+			client, err = metalgo.NewClient(driverURL, metalgo.BearerToken(apiToken), metalgo.HMACAuth(hmacKey, hmacAuthType), metalgo.Transport(transport))
+		}
```

Comment:
Please reduce complexity by creating a list of options.

Reply:
c88156d